registry/client/errors_test.go

```diff
@@ -117,3 +124,17 @@ func TestHandleErrorResponseInsufficientPrivileges403(t *testing.T) {
 		t.Errorf("Expected \"%s\", got: \"%s\"", expectedMsg, err.Error())
 	}
 }
+
+func TestHandleErrorResponseNonJson(t *testing.T) {
+	msg := `{"details":"requesting higher privileges than access token allows"}`
+	response := &http.Response{
+		Status:     "403 Forbidden",
+		StatusCode: 403,
+		Body:       nopCloser{bytes.NewBufferString(msg)},
+	}
+	err := HandleErrorResponse(response)
+
+	if !strings.Contains(err.Error(), msg) {
+		t.Errorf("Expected \"%s\", got: \"%s\"", msg, err.Error())
```

Comment:
Perhaps use `%q` here. But if that makes the output harder to read, you can use string-literal (backtick), and `%v` for the error;

```suggestion
		t.Errorf(`Expected "%s", got: "%v"`, msg, err)
```

Reply:
I kept the original code as is, but yeah we should do some cleanup

Reply:
I've formatted them all with `%q`; no need to make drastic changes in here, imho